test/options_test.cc

```diff
@@ -272,10 +273,115 @@ TEST_F(OptionsImplTest, RequestSource) {
   // Check that our conversion to CommandLineOptionsPtr makes sense.
   CommandLineOptionsPtr cmd = options->toCommandLineOptions();
   EXPECT_EQ(cmd->request_source().uri(), request_source);
+  // TODO(#433)
   OptionsImpl options_from_proto(*cmd);
   EXPECT_TRUE(util(*(options_from_proto.toCommandLineOptions()), *cmd));
 }
 
+class RequestSourcePluginTestFixture : public OptionsImplTest,
+                                       public WithParamInterface<std::string> {};
+
+TEST_P(RequestSourcePluginTestFixture, CreatesOptionsImplWithRequestSourceConfig) {
+  Envoy::MessageUtil util;
+  const std::string request_source_config = GetParam();
+  std::unique_ptr<OptionsImpl> options = TestUtility::createOptionsImpl(
+      fmt::format("{} --request-source-plugin-config {} {}", client_name_, request_source_config,
+                  good_test_uri_));
+
+  CommandLineOptionsPtr command = options->toCommandLineOptions();
+  EXPECT_TRUE(
+      util(command->request_source_plugin_config(), options->requestSourcePluginConfig().value()));
+  // Now we construct a new options from the proto we created above. This should result in an
+  // OptionsImpl instance equivalent to options. We test that by converting both to yaml strings,
+  // expecting them to be equal. This should provide helpful output when the test fails by showing
+  // the unexpected (yaml) diff.
+
+  // The predicates are defined as proto maps, and these seem to re-serialize into a different
+  // order. Hence we trim the maps to contain a single entry so they don't thwart our textual
+  // comparison below.
+  EXPECT_EQ(1, command->mutable_failure_predicates()->erase("benchmark.http_4xx"));
+  EXPECT_EQ(1, command->mutable_failure_predicates()->erase("benchmark.http_5xx"));
+  EXPECT_EQ(1, command->mutable_failure_predicates()->erase("requestsource.upstream_rq_5xx"));
+  // TODO(#433)
+  OptionsImpl options_from_proto(*command);
+  std::string s1 = Envoy::MessageUtil::getYamlStringFromMessage(
+      *(options_from_proto.toCommandLineOptions()), true, true);
+  std::string s2 = Envoy::MessageUtil::getYamlStringFromMessage(*command, true, true);
```

Comment:
One thing that might make this easier to parse is if we use more descriptive names, rather than s1 and s2.  Maybe expected_options_yaml and actual_options_yaml?  Or whatever names you think would be accurate/helpful.

Reply:
For sure.